tests/simulation/CMakeLists.txt

```diff
@@ -28,9 +28,35 @@ function(add_simulation_test test_cpp)
   add_dependencies(memgraph__simulation ${target_name})
 endfunction(add_simulation_test)
 
-add_simulation_test(basic_request.cpp address)
-add_simulation_test(raft.cpp address)
-add_simulation_test(trial_query_storage/query_storage_test.cpp address)
-add_simulation_test(sharded_map.cpp address)
-add_simulation_test(shard_request_manager.cpp address)
+add_simulation_test(basic_request.cpp)
+add_simulation_test(raft.cpp)
+add_simulation_test(trial_query_storage/query_storage_test.cpp)
+add_simulation_test(sharded_map.cpp)
 add_simulation_test(shard_rsm.cpp)
+
+
+function(add_simulation_property test_cpp)
+  # get exec name (remove extension from the abs path)
+  get_filename_component(exec_name ${test_cpp} NAME_WE)
+  set(target_name ${test_prefix}${exec_name})
+  add_executable(${target_name} ${test_cpp})
+
+  # OUTPUT_NAME sets the real name of a target when it is built and can be
+  # used to help create two targets of the same name even though CMake
+  # requires unique logical target names
+  set_target_properties(${target_name} PROPERTIES OUTPUT_NAME ${exec_name})
+
+  # sanitize
+  target_compile_options(${target_name} PRIVATE -fsanitize=address)
+  target_link_options(${target_name} PRIVATE -fsanitize=address)
```

Comment:
I think using the address sanitizer this way is not correct, but at least misleading. We have a CMake configuration, called `ASAN` to enable the address sanitizer for the whole codebase. On top of linking `libasan`, it will also disable `jemalloc`. As both `jemalloc` and `libasan` defines their own `malloc`/`free`, I think they are not working well together. Probably we will use one of the `malloc`/`free` (I don't know which one), but the two things can easily conflict.

As a result, I think we shouldn't enable ASAN for some subset of libraries/executables, but only globally for everything.